packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -325,10 +326,59 @@ final class SignInStateMachine
         createSmsMfaRequest(event),
       ChallengeNameType.softwareTokenMfa when hasUserResponse =>
         createSoftwareTokenMfaRequest(event),
+      ChallengeNameType.emailOtp when hasUserResponse =>
+        createEmailMfaRequest(event),
       ChallengeNameType.selectMfaType when hasUserResponse =>
         createSelectMfaRequest(event),
-      ChallengeNameType.mfaSetup when hasUserResponse =>
-        createMfaSetupRequest(event),
+      ChallengeNameType.mfaSetup => (() async {
+          final allowedMfaTypes = _allowedMfaTypes;
```

Comment:
nit: Unless we want both the no MFA and no eligible MFA we can reduce the number of branches:

```
// Exclude MfaType.sms from consideration
final mfaTypesForSetup = allowedMfaTypes?.difference({MfaType.sms});
if (mfaTypesForSetup == null || mfaTypesForSetup.isEmpty) {
    throw const InvalidUserPoolConfigurationException(
        'No eligible MFA types are available for setup.',
        recoverySuggestion: 'Check your user pool MFA configuration.',
    );
}